file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,523 @@
+      "! <p class="shorttext">Selection Type</p>
+      "! Selection type
+      selection_type              TYPE ty_selection_type,
+      " (type /aif/msg_tbl)
+      "! <p class="shorttext">Message Index Table</p>
+      "! Message index table
+      message_index_table         TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/field_number_parent)
+      "! <p class="shorttext">Parent Field Sequence Number</p>
+      "! Parent field sequence number
+      parent_field_sequence       TYPE n LENGTH 3,
+      " (type /aif/icon_l4)
+      "! <p class="shorttext">Icon</p>
+      "! Icon
+      icon                        TYPE c LENGTH 4,
+      " (type /aif/tooltip)
+      "! <p class="shorttext">Icon Tooltip</p>
+      "! Icon tooltip
+      icon_tooltip                TYPE c LENGTH 40,
+      " (type /aif/alertfieldname)
+      "! <p class="shorttext">Alert Recipient Field Name</p>
+      "! Field name in alert recipient assignment table
+      alert_field_name            TYPE c LENGTH 45,
+      " (type /aif/alrt_catfn)
+      "! <p class="shorttext">Category Field Name</p>
+      "! Category field name
+      category_field_name         TYPE c LENGTH 32,
+      " /AIF/IS_HIDDEN, swap to positive stature
+      "! <p class="shorttext">Show Tree Node</p>
+      "! Show tree node in view 1 tree
+      "! $default 'X'
+      is_tree_node_visible        TYPE abap_bool,
+      " (type /aif/alertrelevant)
+      "! <p class="shorttext">Relevant for Alert Recipient Determination</p>
+      "! Relevant for alert recipient determination
+      is_alert_recipient_relevant TYPE abap_bool,
+    END OF ty_key_field_selection.
+
+  TYPES:
+    "! <p class="shorttext">SAP Fiori Features</p>
+    "! SAP Fiori specific features
+    BEGIN OF ty_fiori_features,
+      " (type /aif/semantic_object)
+      "! <p class="shorttext">Semantic Object</p>
+      "! Semantic object
+      semantic_object             TYPE zif_aff_types_v1=>ty_object_name_30,
+      " (type /aif/semantic_action)
+      "! <p class="shorttext">Semantic Action</p>
+      "! Semantic action
+      semantic_action             TYPE zif_aff_types_v1=>ty_description_60,
+      " /AIF/NOT_SHOW_FIELD
+      "! <p class="shorttext">Display in Message Monitoring</p>
+      "! Display in message monitoring
+      "! $default 'X'
+      is_shown_in_message_monitor TYPE abap_bool,
+      " /AIF/IS_DEFAULT_FIELD
+      "! <p class="shorttext">Display in Message Monitoring by Default</p>
+      "! Display in message monitoring by default
+      is_shown_by_default         TYPE abap_bool,
+    END OF ty_fiori_features.
+
+  TYPES:
+    " Related key fields used by key field rules
+    "! <p class="shorttext">Rule Key Field</p>
+    "! Rule key field
+    BEGIN OF ty_rule_key_field,
+      " (type /aif/key_fieldname_enh)
+      "! <p class="shorttext">Rule Key Field Name</p>
+      "! Rule Key field name
+      rule_key_field_name  TYPE zif_aff_types_v1=>ty_object_name_30,
```

Comment:
Maybe 
```suggestion
      " (type /aif/key_fieldname_enh)
      "! <p class="shorttext">Name</p>
      "! Rule Key field name
      name  TYPE zif_aff_types_v1=>ty_object_name_30,
```
is enough here. Otherwise, you have in structure Key Field Rule the field Key Field Rule Name which is duplicate info.

Reply:
Honestly I do not really get the point, but I adjusted it. Commit will follow.